components/datetime/src/neo.rs

```diff
@@ -657,7 +697,7 @@ where
     /// [`AnyCalendarKind`]: icu_calendar::AnyCalendarKind
     #[inline(never)]
     #[cfg(feature = "compiled_data")]
-    pub fn try_new(locale: &DataLocale, length: NeoSkeletonLength) -> Result<Self, LoadError>
+    pub fn try_new(locale: &DataLocale, options: impl Into<NeoOptions<R>>) -> Result<Self, LoadError>
```

Comment:
issue: Don't put `impl Into` on APIs, let the caller call `.into()`.

Reply:
Yeah so I'm usually the person who says we shouldn't have `impl Trait` on APIs. I did it here because

1. The functions are already highly generic
2. The `format` function takes an `impl Trait` argument (both in 1.0 and 2.0)
3. The length argument is the only argument people will need to pass in a large majority of cases

I'm willing to remove the `impl Trait`, but which call site would you suggest using in the docs?

```rust
// Current
TypedNeoFormatter::<Gregorian, NeoYearMonthDayMarker>::try_new(
    &locale!("es-MX").into(),
    NeoSkeletonLength::Long,
)

// With .into()
TypedNeoFormatter::<Gregorian, NeoYearMonthDayMarker>::try_new(
    &locale!("es-MX").into(),
    NeoSkeletonLength::Long.into(),
)

// With From
use icu::datetime::NeoOptions;
TypedNeoFormatter::<Gregorian, NeoYearMonthDayMarker>::try_new(
    &locale!("es-MX").into(),
    NeoOptions::from(NeoSkeletonLength::Long),
)

// With struct construction
use icu::datetime::NeoOptions;
TypedNeoFormatter::<Gregorian, NeoYearMonthDayMarker>::try_new(
    &locale!("es-MX").into(),
    {
        let mut options = NeoOptions::default();
        options.length = NeoSkeletonLength::Long;
        options
    }
)
```

Reply:
The second one. We're also doing this for `FallbackOptions` which is constructed as `DeduplicationStrategy::Foo.into()`.

Reply:
Done.